packages/connectivity/ios/Classes/ConnectivityPlugin.m

```diff
@@ -97,6 +103,41 @@ - (NSString*)statusFromReachability:(Reachability*)reachability {
   }
 }
 
+- (void)requestLocationServiceAuthorizationIfUndetermined:(FlutterResult)result always:(NSNumber *)always {
+  CLAuthorizationStatus status = CLLocationManager.authorizationStatus;
+  switch (status) {
+    case kCLAuthorizationStatusNotDetermined: {
+      self.result = result;
```

Comment:
I think there's a race condition here where two calls to `requestLocationServiceAuthorizationIfUndetermined` in quick succession could lead to result being clobbered before being invoked. Consider returning an error if self.result != null